### PR TITLE
IFA: Support for show tam int-ifa statistics command

### PR DESCRIPTION
**Pull request by spenugondaa**
```
sonic# show tam int-ifa statisitcs
  String(Max: 32 characters)  Name of the flow
  |                           Pipe through a command
  <cr>

sonic# show tam int-ifa statisitcs
------------------------------------------------------------------------------------------------
FLOW           ACL TABLE      ACL RULE       PACKET COUNT   BYTE COUNT
------------------------------------------------------------------------------------------------
c1             acl1           rule1          0              0
c2             acl2           rule2          0              0
sonic# show tam int-ifa statisitcs all
------------------------------------------------------------------------------------------------
FLOW           ACL TABLE      ACL RULE       PACKET COUNT   BYTE COUNT
------------------------------------------------------------------------------------------------
c1             acl1           rule1          0              0
c2             acl2           rule2          0              0
sonic# show tam int-ifa statisitcs c1
------------------------------------------------------------------------------------------------
FLOW           ACL TABLE      ACL RULE       PACKET COUNT   BYTE COUNT
------------------------------------------------------------------------------------------------
c1             acl1           rule1          0              0
sonic# show tam int-ifa statisitcs c2
------------------------------------------------------------------------------------------------
FLOW           ACL TABLE      ACL RULE       PACKET COUNT   BYTE COUNT
------------------------------------------------------------------------------------------------
c2             acl2           rule2          0              0

```

### CODE REVIEW

File: src/CLI/actioner/sonic-cli-ifa.py [hunk 1]
```diff
@@ -132,11 +132,42 @@ def get_tam_ifa_status(args):
...
+
+    if response.ok():
+        if response.content:
+            if len(args) == 0:
+                api_response = response.content['sonic-ifa:TAM_INT_IFA_FLOW_TABLE']['TAM_INT_IFA_FLOW_TABLE_LIST']
+            else:
+                api_response = response.content['sonic-ifa:TAM_INT_IFA_FLOW_TABLE_LIST']
+            for i in range(len(api_response)):
+                acl_counter_key = 'COUNTERS:' + api_response[i]['acl-table-name'] + ':' + api_response[i]['acl-rule-name']
+                flow_stats = counters_db.get_all(counters_db.COUNTERS_DB, acl_counter_key)
```
Comment (asha-behera) [inline src/CLI/actioner/sonic-cli-ifa.py]: Should we use data from Redis-DB? Can't we use REST for the same ?

Comment (asha-behera) [reply src/CLI/actioner/sonic-cli-ifa.py]: How the user will be able to get statistics data for REST?

Comment (spenugondaa) [reply src/CLI/actioner/sonic-cli-ifa.py]: @asha-behera Right now the support is only for CLI. 
As per @PrabhuSreenivasan the counters are available (looks to be only available) at counters DB.  Sonic-acl.yang doesn’t contain any stats container for me to use rest API to fetch them in my actioner.

For supporting ifa stats command through rest api: is there a backend support to populate a counters DB table by IFA? If so, I could define a stats container in ifa yang.

Comment (asha-behera) [reply src/CLI/actioner/sonic-cli-ifa.py]: @spenugondaa : stats info should be "state" ( config=false)  information and not config information.

Comment (spenugondaa) [reply src/CLI/actioner/sonic-cli-ifa.py]: @asha-behera @PrabhuSreenivasan please take a look. I am using now oc acl yang to fetch the counters.

File: src/CLI/actioner/sonic-cli-ifa.py [hunk 2]
```diff
@@ -153,11 +149,22 @@ def get_tam_ifa_flow_stats(args):
...
-                flow_stats = counters_db.get_all(counters_db.COUNTERS_DB, acl_counter_key)
-                api_response[i]['Packets'] = flow_stats['Packets']
-                api_response[i]['Bytes'] = flow_stats['Bytes']
+                rule = api_response[i]['acl-rule-name']
+                print(rule)
+                # tokenize the rulename with '_' and fetch last number
+                tmpseq = (rule.split("_", 1))[-1]
+                print(tmpseq)
+                path = cc.Path('/restconf/data/openconfig-acl:acl/acl-sets/acl-set={name},{type}/acl-entries/acl-entry={seqid}', name=api_response[i]['acl-table-name'], type="ACL_IPV4", seqid=tmpseq)
+                flow_stats = api.get(path)
```
Comment (asha-behera) [inline src/CLI/actioner/sonic-cli-ifa.py]: We should not be hard-coding type...it can be ipv6 also

Comment (spenugondaa) [reply src/CLI/actioner/sonic-cli-ifa.py]: @asha-behera I did thought about it. But a flow only contains acl table/rule and a collector. Should I read collector properties and decide if I have to use v4 or v6 with acl type? Which doesn't sound right.. as a collector is just a receiver.

Comment (asha-behera) [reply src/CLI/actioner/sonic-cli-ifa.py]: The type should be the type of ACL, I think IP_TYPE field has been added in OC ACL to know the type of ACL. I am not sure if the change has been merged.  I not then we can use ACLNAME which would have IPV4 or IPV6 appended to it like (..e.g TestACL100_ACL_IPV4)

Comment (spenugondaa) [reply src/CLI/actioner/sonic-cli-ifa.py]: @asha-behera Using get_openconfig_acl_acl_acl_sets() to fetch acl list to index into the table/rule to get counters.
Please approve so we can merge.

**Review by asha-behera — APPROVED** [review]

**Pull request merged**